File: Day 10 : Binary Numbers/Solution.py
if __name__ == '__main__':
    n = int(input().strip())
    count, no_of_1s = 0, 0
    
    while(n > 0):
        if(n % 2 == 1):
            count += 1
            if(count > no_of_1s):
                no_of_1s = count
        else:
            count = 0
        
        n //= 2
    
    print(no_of_1s)

File: Day 10 : Binary Numbers/Solution.rb
n = gets.strip.to_i

# count of the or bits
max = 0
count = 0

# divide and store the remainder as bit, quotient as n
while n > 0
    bit = n % 2
    if bit == 1
        count += 1
        if count > max
            max = count
        end
    else
        count = 0
    end
    n = (n / 2).floor() 
end

puts max
